### ISSUE TITLE
New JavaScript Project

### ISSUE THREAD

**Issue by boniatillo-com** [angelozerr/typescript.java#200]
Hi,

It would be nice to have a New JavaScript Project that generates a project with a jsconfig.json file. 

TypeScript IDE is good also for JavaScript development.

**Comment by angelozerr** [comment]
Yes sure it's a good idea. We could use some code of the new typscript project wizard.

Perhaps adds some other config like typing?

Any idea and contribution are welcome!

**Comment by boniatillo-com** [comment]
I don't know what you mean with "config typing", the types definitions files foo.d.ts? Maybe some parameters from here, like `target` and `module`:

https://code.visualstudio.com/docs/languages/jsconfig

**Comment by boniatillo-com** [comment]
I was playing a bit with TypeScript IDE and for TypeScript it is working very nice, however it is obvious that it needs to give some love to JavaScript. I think this tool could be a great JavaScript IDE, something that in my opinion still an open issue in the Eclipse community.

**Comment by angelozerr** [comment]
> I don't know what you mean with "config typing"

I mean https://github.com/typings/typings to download jquery definition for instance but ATA can do that too.

> I think this tool could be a great JavaScript IDE, something that in my opinion still an open issue in the Eclipse community.

Thanks a lot @boniatillo-com! I think typescript.java starts be mature and I'm studying if I could move my work to https://github.com/mickaelistria/eclipse-bluesky which uses tm4e, lsp4e. IMHO I think it's the best option, but lsp4e and GenericEditor must be improved again to support the same features than typescript.java (matching brackets, compile on save, support for lsp4e with tsserver etc)

**Comment by boniatillo-com** [comment]
Yes, I think Eclipse is going to enter to the modern age with the help of Generic Editor, TM4E and LSP4E.
I agree with you that typescript.java is getting mature, I am exciting with the idea of provide TypeScript IDE by default in Phaser Editor, it is going to happen soon.

That's great that you want to collaborate with eclipse bluesky, I have the faith that Eclipse will get relevance again in the non Java world.
